cmd/deploy.go

```diff
@@ -216,6 +216,7 @@ func (dc *deployCmd) loadAPIModel(cmd *cobra.Command, args []string) error {
 	}
 
 	if dc.containerService.Properties.IsAzureStackCloud() {
+		dc.containerService.SetCustomCloudProfileEnvironment()
```

Comment:
what about using the `generate` + `az deploy` way of deploying (https://github.com/Azure/aks-engine/blob/master/docs/tutorials/deploy.md#aks-engine-the-long-way) 

would it make sense in general to move this function outside of commands and into template generation since deploy, generate, upgrade and scale all call it?

Reply:
For generate and deploy, it is implicitly invoked by SetPropertiesDefaults.

SetCustomCloudProfileEnvironment() is to set the default value and invoked by all 4 commands.
this is the method with this pattern. I am open for either way. 
@jackfrancis what do you think?

Reply:
I don't see `SetCustomCloudProfileEnvironment` being called in `deploy.go`. Are you saying that it's already invoked in `SetPropertiesDefault`?

Reply:
yes, currently it is correct SetCustomCloudProfileEnvironment is invoked in SetPropertiesDefault.

for deploy command, SetCustomCloudProfileEnvironment need to be invoked before get getClient,  
I could not find easy way to move SetPropertiesDefault before getClient, I explicitly invoke  SetCustomCloudProfileEnvironment before getClient

so SetCustomCloudProfileEnvironment will be invoked twice for deploy command.
the second call SetCustomCloudProfileEnvironment will not make any changes since all required values are set.

So we have two options here

1. remove SetCustomCloudProfileEnvironment from SetPropertiesDefault and call SetCustomCloudProfileEnvironment for generate and deploy command.

2. (the current code) call  SetCustomCloudProfileEnvironment inside SetPropertiesDefault and only call SetCustomCloudProfileEnvironment for deploy command.

Both way are ok for me.
@CecileRobertMichon please let me know your preference.

Reply:
Ok makes sense, thanks for the explanation. I think the way you're doing it now is fine in that case, unless there's a way to call getClient after `SetPropertiesDefault` (but looks like you tried that and it wasn't straightforward.)